backend/app/routers/authorization.py

```diff
@@ -0,0 +1,55 @@
+from fastapi import APIRouter, Depends
+from pymongo import MongoClient
+
+from app import keycloak_auth, dependencies
+from app.deps.authorization_deps import Authorization, get_role
+from app.keycloak_auth import get_current_username
+from app.models.authorization import AuthorizationBase, AuthorizationDB, RoleType
+from bson.objectid import ObjectId as BsonObjectId
+
+from app.models.pyobjectid import PyObjectId
+
+router = APIRouter()
+
+
+@router.post("", response_model=AuthorizationDB)
+async def save_authorization(
+    authorization_in: AuthorizationBase,
+    user=Depends(keycloak_auth.get_current_username),
+    db: MongoClient = Depends(dependencies.get_db),
+):
+    """Save authorization info in Mongo. This is a triple of dataset_id/user_id/role."""
+    authorization_db = AuthorizationDB(**authorization_in.dict(), creator=user)
+    new_authorization = await db["authorization"].insert_one(
```

Comment:
Should we check if user exist first before insert?

Reply:
Not sure. `get_current_username` gets the user id from the token, so in theory it should exist, since the token was created. That method throws an exceptions if there is a KeycloakAuthenticationError. We could also use `get_current_user` which goes to Mongo, but I am not sure that is necessary.